client/src/components/LoginPage/index.js

```diff
@@ -1,86 +1,110 @@
-        password: user["password"],
-      });
+    if (name === "email") {
+      setTempUser({ ...temp, email: value });
+      setEmailError(false);
     } else if (name === "password") {
-      setTempUser({
-        username: user["username"],
-        email: user["email"],
-        password: value,
-      });
+      setTempUser({ ...temp, password: value });
+      setPasswordError(false);
     }
   };
 
-  function submitForm() {
-    console.log("Submitting");
-    const resp = validateUser();
+  // does a GET /users call with temp's saved credentials
+  async function getUser(emailID, passwordID) {
+    try {
+      const resp = await axios.get("http://127.0.0.1:5000/users", {
+        auth: {
+          username: emailID,
+          password: passwordID,
+        },
+      });
 
-    if (resp === 0) {
+      // Query went through successfully
+      return resp;
+    } catch (e) {
+      console.log("Log in failed");
+      return e;
+    }
+  }
+
+  // when user submits to login
+  async function submitForm() {
+    console.log("Submitting...");
+    var resp = await validateUser();
+    // successful GET /users from backend
+    // set user and set route to main listing page
+    if (resp === 200) {
       console.log("Successful Login credentials");
       props.history.push("/browse");
-    } else if (resp === -1) {
-      alert("Invalid login");
+      setUser(temp);
+    }
+    // failed GET /users from backend
+    // means incorrect credentials - Try again
+    else if (resp === 401) {
+      console.log("Invalid Credentials");
+      setError(true);
+    }
+    // a field was left empty
+    else if (resp === -1) {
+      console.log("Empty Field(s)");
+      setError(true);
     }
   }
 
-  function validateUser() {
-    // potentially change to allow username OR email
-    // change to check database
-    if (
-      user.username === "DummyUser" &&
-      user.email === "Dummy@yahoo.com" &&
-      user.password == "Password123"
-    )
-      return 0;
-
-    if (user.username.length === 0) {
-      alert("Please enter a Username");
-      return -2;
-    } else if (user.email.length === 0) {
-      alert("Please enter an Email");
-      return -2;
-    } else if (user.password.length === 0) {
-      alert("Please enter a Password");
-      return -2;
+  // references credentials with backend.
+  // resp = 200 if successful, 401 if incorrect, and -1 if any are empty
+  // any empty fields turn red
+  async function validateUser() {
+    let res = 0;
+    if (temp.email.length === 0) {
+      setEmailError(true);
+      res = -1;
+    }
+    if (temp.password.length === 0) {
+      setPasswordError(true);
+      res = -1;
+    }
+    if (res === -1) {
+      setError(true);
+      return -1;
+    }
+    // if both fields have something then check backend
+    // if resp.status === 200 then success
+    // if resp.status === 'undefined', then it's an error and return resp.response.status (401)
+    else {
```

Comment:
I would consider adding this logic to your submitForm() function.  This could almost replace the second part of your submitForm function, if you did setUser(resp.data).